.github/workflows/extra_tests.yml

```diff
@@ -63,10 +70,36 @@ jobs:
           key: cargo-registry-${{ hashFiles('shadow/src/Cargo.lock') }}
           restore-keys: |
             cargo-registry-
+      - name: Restore sccache cache
+        uses: actions/cache@v3
+        continue-on-error: false
+        with:
+          path: ${{ env.SCCACHE_DIR }}
+          key: sccache-${{ env.CONTAINER }}-${{ env.CC }}-${{ env.BUILDTYPE }}-${{ hashFiles('shadow/src/Cargo.lock') }}
```

Comment:
I think this should have the rust compiler version, since if the rust version changes everything needs to be rebuilt and we should invalidate the cache

Reply:
done